### PR TITLE
Feature/agent

### PR DESCRIPTION
**Pull request by ketd**
# Pull Request - GTPlanner 核心功能增强

## 📋 Description

**Summary of changes:**
本 PR 包含了 GTPlanner 项目的重大功能增强和架构优化，主要涵盖多语言支持、CLI 改进、流式响应、ReAct 系统集成等核心功能的实现和完善。

**Related issue(s):**
- 实现多语言国际化支持
- 优化 CLI 用户体验
- 增强 API 流式响应能力
- 完善 ReAct 智能控制系统

**Motivation and context:**
为了提升 GTPlanner 作为智能任务规划系统的用户体验和功能完整性，本次更新重点解决了以下问题：
1. 缺乏多语言支持导致的国际化障碍
2. CLI 界面需要更好的用户交互体验
3. API 需要支持实时流式响应
4. 系统架构需要更好的模块化和可扩展性

## 🔄 Type of Change

Please mark the relevant option(s):

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [x] 🔧 Code refactoring (no functional changes)
- [x] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Code style/formatting changes

## 🛠️ Changes Made

**主要功能模块:**

### 🌐 多语言支持 (commits: fdab87a, 123b2b6, a60b7df)
- `agent/cli/cli_text_manager.py` - 新增 CLI 多语言文本管理器
- `agent/cli/gtplanner_cli.py` - 重构 CLI 主文件，集成多语言支持
- `README.md` & `README_zh-CN.md` - 更新项目文档和贡献指南
- `ASCll.txt` - 添加 ASCII 艺术 logo

### 🔄 流式响应系统 (commits: 2a75068, 06c01c9, cd568e4)
- `agent/streaming/event_helpers.py` - 增强事件处理机制
- 实现 SSE (Server-Sent Events) 流式响应支持
- 在 assistant_message_end 事件中包含 tool_calls 信息
- 重构文件保存为事件驱动模式

### 🏗️ 架构重构 (commits: 81e3b33, c0a38f4, 9adbdcd, 0e6749a)
- 重构设计流程架构并添加新功能
- 实现状态机工具调用标签过滤器
- 完善系统架构和功能模块
- 优化 CLI、工具系统和数据持久化模块

### 🤖 ReAct 系统集成 (commits: 79a7a6c, e7d31bb, ae44639)
- 完善 ReAct 系统集成和相关组件更新
- 实现完全异步的 ReAct 智能主控制器系统
- 完成核心架构重构 - SQLite 集成和智能压缩系统

### 🔧 工具系统优化 (commits: e607217, 327ae0f, 3fdbe57)
- 增加工具创建(init)节点和工具推荐节点
- 更新 agent 相关功能和调试工具
- 剔除冗余的需求分析工具

**Key changes:**
- **多语言支持**: 实现了完整的 CLI 多语言文本管理系统
- **流式响应**: 添加了 SSE 支持，提升实时交互体验
- **架构优化**: 重构了核心架构，提高了系统的可维护性和扩展性
- **ReAct 集成**: 完善了智能控制系统，增强了自动化能力
- **工具系统**: 优化了工具推荐和创建流程

## 🧪 Testing Performed

**Manual testing completed:**
- [x] CLI 多语言功能测试
- [x] 流式响应 API 测试
- [x] ReAct 系统集成测试
- [x] 工具创建和推荐功能测试
- [x] 错误处理和边界情况测试

**Test commands used:**
```bash
# CLI 多语言测试
uv run python gtplanner.py --help
uv run python gtplanner.py --language zh-CN

# 流式响应测试
curl -X POST http://localhost:8000/api/stream \
  -H "Content-Type: application/json" \
  -d '{"message": "test"}'

# ReAct 系统测试
uv run python -m agent.react_controller

# 工具系统测试
uv run python -m agent.tools.tool_recommender
```

**Test scenarios covered:**
1. **多语言切换**: 测试中英文界面切换，确保文本正确显示
2. **流式响应**: 验证 SSE 连接稳定性和数据完整性
3. **ReAct 控制**: 测试智能控制器的决策和执行能力
4. **工具推荐**: 验证工具推荐算法的准确性和相关性

**Sample input/output:**
```
Input: "gtplanner --language zh-CN --help"
Output: "GTPlanner 智能任务规划系统
使用方法: gtplanner [选项] [命令]
..."
```

## 🌍 Environment Tested

- **Python version**: 3.10+
- **Operating System**: macOS, Linux, Windows
- **Package manager**: uv
- **LLM Provider**: OpenAI, Anthropic
- **Database**: SQLite (内置)

## 💥 Breaking Changes

**Are there any breaking changes?**
- [ ] No breaking changes
- [x] Yes, breaking changes (describe below)

**If yes, describe the breaking changes:**
- **CLI 重命名**: `modern_gtplanner_cli.py` 重命名为 `gtplanner_cli.py`
- **配置文件**: 多语言配置可能需要重新设置
- **API 接口**: 部分流式响应接口的返回格式有调整

**Migration guide:**
1. 更新 CLI 调用方式：使用新的 `gtplanner_cli.py`
2. 检查配置文件中的语言设置
3. 更新 API 客户端以适配新的流式响应格式

## 📚 Documentation

- [x] Code comments added/updated
- [x] README.md updated
- [x] CONTRIBUTING.md updated
- [ ] API documentation updated (if applicable)
- [ ] No documentation changes needed

**Documentation updates:**
- 更新了中英文 README 文件
- 添加了贡献指南
- 完善了代码注释和文档字符串

## 📦 Dependencies

**New dependencies added:**
- [ ] No new dependencies
- [x] New dependencies added (list below)

**If new dependencies were added:**
- 主要依赖于现有的 Python 标准库和已有依赖
- 多语言支持使用内置的 JSON 配置文件
- 流式响应基于现有的异步框架

## ⚡ Performance Impact

- [ ] No performance impact
- [x] Performance improvement
- [ ] Potential performance regression (explain below)

**Performance notes:**
- **流式响应**: 显著提升了实时交互的响应速度
- **异步架构**: ReAct 系统的异步实现提高了并发处理能力
- **智能压缩**: SQLite 集成和智能压缩系统优化了存储效率
- **模块化设计**: 减少了不必要的模块加载，提升启动速度

## 🔒 Security Considerations

- [x] No security implications
- [ ] Security improvements
- [ ] Potential security concerns (explain below)

**Security notes:**
本次更新主要涉及功能增强和架构优化，未引入新的安全风险。所有用户输入都经过适当的验证和处理。

## ✅ Reviewer Checklist

**For reviewers to complete:**

**Code Quality:**
- [ ] Code follows project style guidelines
- [ ] Code is well-documented and readable
- [ ] Error handling is appropriate
- [ ] No obvious bugs or issues

**Functionality:**
- [ ] Changes work as described
- [ ] No regressions in existing functionality
- [ ] Edge cases are handled appropriately
- [ ] Integration points work correctly

**Testing:**
- [ ] Manual testing is adequate
- [ ] Test scenarios cover the changes
- [ ] Error conditions are tested
- [ ] Performance is acceptable

**Documentation:**
- [ ] Code is properly documented
- [ ] User-facing documentation is updated
- [ ] Breaking changes are clearly documented

## 📝 Additional Notes

**Anything else reviewers should know:**
- 本次 PR 包含了多个重要功能的集成，建议分模块进行详细审查
- 多语言支持的文本配置文件需要特别关注翻译质量
- 流式响应功能需要在不同网络环境下进行测试
- ReAct 系统的异步实现需要关注并发安全性

**Follow-up tasks:**
- [ ] 添加更多语言支持（如日语、韩语等）
- [ ] 完善 API 文档和示例
- [ ] 增加更多单元测试和集成测试
- [ ] 优化错误处理和日志记录

---

**Checklist for submitter:**
- [x] I have tested these changes thoroughly
- [x] I have updated documentation as needed
- [x] I have followed the project's coding standards
- [x] I have checked for any breaking changes
- [x] I have linked related issues
- [x] I am ready for code review

## 📊 Commit Summary

本 PR 包含以下 23 个提交，时间跨度从 2025-08-06 到 2025-08-26：

```
fdab87a - 更新readme和贡献指南 (ketd, 2025-08-26)
123b2b6 - 添加cil多语言支持 (ketd, 2025-08-26)
a60b7df - 添加多语言提示词模板功能 (ketd, 2025-08-26)
2a75068 - 将文件保存重构为通过事件来处理 (ketd, 2025-08-22)
06c01c9 - feat: 添加 SSE 流式响应支持和增强的 API 功能 (ketd, 2025-08-21)
cd568e4 - feat: 在 assistant_message_end 事件中包含 tool_calls 信息 (ketd, 2025-08-21)
81e3b33 - refactor: 重构设计流程架构并添加新功能 (ketd, 2025-08-19)
c0a38f4 - feat: 实现状态机工具调用标签过滤器 (ketd, 2025-08-19)
9adbdcd - feat: 完善系统架构和功能模块 (ketd, 2025-08-15)
0e6749a - 重构代码结构：优化CLI、工具系统和数据持久化模块 (ketd, 2025-08-14)
769873e - merge: 合并feature/add_tools_recommend分支 (ketd, 2025-08-14)
327ae0f - feat: 更新agent相关功能和调试工具 (ketd, 2025-08-14)
ae44639 - feat: 完成核心架构重构 - SQLite集成和智能压缩系统 (ketd, 2025-08-13)
6080327 - 优化提示词 (ketd, 2025-08-13)
22fbd02 - 代码优化 (ketd, 2025-08-12)
7a05a9a - 添加tracing (ketd, 2025-08-11)
3fdbe57 - 剔除需求分析工具 (ketd, 2025-08-11)
19e2e3e - 🔧 修复工具调用和状态管理关键问题 (ketd, 2025-08-08)
54050a1 - 版本暂存，修改为异步节点模式 (ketd, 2025-08-07)
933862f - 版本暂存 (ketd, 2025-08-07)
79a7a6c - feat: 完善ReAct系统集成和相关组件更新 (ketd, 2025-08-06)
e7d31bb - feat: 实现完全异步的ReAct智能主控制器系统 (ketd, 2025-08-06)
e607217 - 增加工具创建(init)节点,工具推荐节点 (wangyadong, 2025-08-06)
9c42aa2 - feat: 完成Architecture Agent重构和优化 (ketd, 2025-08-06)
```

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,25 @@
+# GTPlanner Configuration
 LLM_API_KEY=your_api_key_here
 LLM_MODEL=deepseek-v3
+JINA_API_KEY=your_jina_api_key_here
```
Comment (redreamality) [inline .env.example:4]: 如果不填会怎样？

Comment (ketd) [reply .env.example:4]: 搜索工具调用失败，我修改一下吧，改为不填的话search工具就不起用

File: README.md [hunk 2]
```diff
@@ -1,58 +1,65 @@
-# GTPlanner: AI-Powered PRD Generation Tool
+# GTPlanner: Intelligent Task Planning System
 
 <p align="center">
   <img src="./assets/banner.png" width="800" alt="GTPlanner Banner"/>
 </p>
 
 <p align="center">
-  <strong>An intelligent Product Requirement Document (PRD) generation tool that transforms natural language descriptions into structured technical documents suitable for Vibe coding.</strong>
+  <strong>GTPlanner (Graph Task Planner) is an intelligent task planning system designed for modern AI-assisted development that automatically generates structured task flowcharts and related documentation based on user requirements.</strong>
```
[the remark this answers is not in the document]
Comment (ketd) [reply README.md:8]: 好的

File: .env.example [hunk 3]
```diff
@@ -1,2 +1,27 @@
+# GTPlanner Configuration
+LLM_BASE_URL=your_llm_base_url_here
 LLM_API_KEY=your_api_key_here
 LLM_MODEL=deepseek-v3
+#非必填，不提供则不会启用搜索工具
```
Comment (redreamality) [inline .env.example:5]: eng

**Review by redreamality — APPROVED** [review]
